python/ray/libraylet_library_python.pyx

```diff
@@ -0,0 +1,723 @@
+        return self.task_spec.get().ParentCounter()
+
+    def function_descriptor_list(self):
+        """Return the function descriptor for this task."""
+        cdef c_vector[c_string] function_descriptor = self.task_spec.get().FunctionDescriptor()
+        results = []
+        for i in range(function_descriptor.size()):
+            results.append(function_descriptor[i])
+        return results
+
+    def arguments(self):
+        """Return the arguments for the task."""
+        cdef:
+            RayletTaskSpecification *task_spec = self.task_spec.get()
+            c_int64 num_args = task_spec.NumArgs()
+            int count
+        arg_list = []
+        for i in range(num_args):
+            count = task_spec.ArgIdCount(i)
+            if count > 0:
+                assert count == 1
+                arg_list.append(PyObjectID.from_native(task_spec.ArgId(i, 0)))
+            else:
+                serialized_str = task_spec.ArgVal(i)[:task_spec.ArgValLength(i)]
+                obj = pickle.loads(serialized_str)
+                arg_list.append(obj)
+        return arg_list
+
+    def returns(self):
+        """Return the object IDs for the return values of the task."""
+        cdef RayletTaskSpecification *task_spec = self.task_spec.get()
+        return_id_list = []
+        for i in range(task_spec.NumReturns()):
+            return_id_list.append(PyObjectID.from_native(task_spec.ReturnId(i)))
+        return return_id_list
+
+    def required_resources(self):
+        """Return the resource vector of the task."""
+        cdef:
+            unordered_map[c_string, double] resource_map = self.task_spec.get().GetRequiredResources().GetResourceMap()
+            c_string resource_name
+            double resource_value
+            unordered_map[c_string, double].iterator iterator = resource_map.begin()
+
+        required_resources = {}
+        while iterator != resource_map.end():
+            resource_name = dereference(iterator).first
+            py_resource_name = str(resource_name)  # bytes for Py2, unicode for Py3
+            resource_value = dereference(iterator).second
+            required_resources[py_resource_name] = resource_value
+            postincrement(iterator)
+        return required_resources
+
+    def actor_creation_id(self):
+        """Return the actor creation ID for the task."""
+        return PyActorID.from_native(self.task_spec.get().ActorCreationId())
+
+    def actor_creation_dummy_object_id(self):
+        """Return the actor creation dummy object ID for the task."""
+        return PyObjectID.from_native(self.task_spec.get().ActorCreationDummyObjectId())
+
+    def actor_id(self):
+        """Return the actor ID for this task."""
+        return PyActorID.from_native(self.task_spec.get().ActorId())
+
+    def actor_counter(self):
+        """Return the actor counter for this task."""
+        return self.task_spec.get().ActorCounter()
+
+    def _serialized_raylet_task(self):
```

Comment:
How do you feel about leaving this in C++ and just wrapping it in Cython? Not sure if it makes a lot of sense to have all these Cython wrappers for flatbuffers.

Reply:
also then we can re-use it in different language implementations in the future